lib/presentation/pages/login_page/bloc/login_bloc.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:injectable/injectable.dart';
+import 'package:trip_tally/domain/entities/user/login_entity.dart';
+import 'package:trip_tally/domain/use_case/login_use_case.dart';
+import 'package:trip_tally/presentation/utils/enums/errors.dart';
+
+part 'login_bloc.freezed.dart';
+
+part 'login_event.dart';
+
+part 'login_state.dart';
+
+@injectable
+class LoginBloc extends Bloc<LoginEvent, LoginState> {
+  LoginBloc(this._loginUseCase) : super(const LoginState.initial()) {
+    on<OnTapLoginEvent>(_onTapLogin);
+  }
+
+  final LoginUseCase _loginUseCase;
+
+  Future<void> _onTapLogin(OnTapLoginEvent event, Emitter<LoginState> emit) async {
+    final login = await _loginUseCase.call(LoginEntity(email: event.email, password: event.password));
+    login.fold(
+      (l) {
+        emit(
+          const LoginState.failure(Errors.somethingWentWrong),
+        );
+        emit(
+          const LoginState.initial(),
+        );
+      },
+      (r) {
+        emit(
+          const LoginState.loading(),
```

Comment:
With the current implementation loading will occur after you got the response from the server and it's success. Loading states should be emitted while you wait for that response. So maybe move the loading state at the beggining of the function